Add CEReactions tests for HTMLButtonElement

xiaoyu3x opened the pull request:
- Verify attributes of HTMLButtonElement interface have CEReactions.
- Adjust functions ‘testReflectAttribute’ and ‘testReflectAttributeWithContentValues’  in the reactions.js to compatible customized built-in element

<!-- Reviewable:start -->

<!-- Reviewable:end -->

xiaoyu3x committed:
Add CEReactions tests for HTMLButtonElement

- Verify attributes of HTMLButtonElement interface have CEReactions
- Adjust functions "testReflectAttribute" and "testReflectAttributeWithContentValues"
in the reactions.js to compatible customized built-in element

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/26327)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/356725793)

Honry committed:
Address @rniwa's comment and fix nit issue

- Expose and rename "testReflectAttrWithContentValuesAndParentNode" directly
- Fix nit: s/formNoValidate/formnovalidate

Honry commented on custom-elements/reactions/HTMLButtonElement.html:
```diff
@@ -0,0 +1,112 @@
+
+//In parent node, sub node's observeAttribute can enqueue by changing attribute value 
+//Test reflect attribute with content values and parent node
+function testReflectAttrWithContentValuesAndPNode(jsAtName, coAtName, jsAtValue1, coAtValue1, jsAtValue2, coAtValue2, name, elementName, pElementName, interfaceName) {
```
@rniwa, @xiaoyu3x is no long active in w-p-t, I just push a new commit that addresses your comment. PTAL.

rniwa commented on custom-elements/reactions/resources/reactions.js:
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
     test(function () {
-        var element = define_new_custom_element([contentAttributeName]);
-        var instance = document.createElement(element.name);
+        if(elementName === undefined) {
```
Nit: should have a space between `if` and `(`.

Honry replied:
@rniwa, thanks for catching this, done.

rniwa commented on custom-elements/reactions/resources/reactions.js:
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
         assert_array_equals(logEntries.types(), ['attributeChanged']);
 
-        assert_attribute_log_entry(logEntries.last(), {name: contentAttributeName, oldValue: null, newValue: contentValue1, namespace: null});
+        assert_attribute_log_entry(logEntries.last(), { name: contentAttributeName, oldValue: null, newValue: contentValue1, namespace: null });
```
Let's not needlessly modify the coding style of the test harness code.

Honry replied:
Fixed.

rniwa commented on custom-elements/reactions/resources/reactions.js:
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
     test(function () {
-        var element = define_new_custom_element([contentAttributeName]);
-        var instance = document.createElement(element.name);
+        if(elementName === undefined) {
```
Ditto.

rniwa commented on custom-elements/reactions/resources/reactions.js:
```diff
@@ -126,32 +126,42 @@ function testCloner(testFunction, name) {
         var logEntries = element.takeLog();
         assert_array_equals(logEntries.types(), ['attributeChanged']);
-        assert_attribute_log_entry(logEntries.last(), {name: contentAttributeName, oldValue: contentValue1, newValue: contentValue2, namespace: null});
+        assert_attribute_log_entry(logEntries.last(), { name: contentAttributeName, oldValue: contentValue1, newValue: contentValue2, namespace: null });
```
Ditto.

Honry committed:
Fix nits

rniwa reviewed (approved):
Great. Thanks for the test!
